model/AssessmentResultsService.php

```diff
@@ -0,0 +1,217 @@
+     */
+    public function getDeliveryData(\taoDelivery_models_classes_execution_DeliveryExecution $deliveryExecution)
+    {
+        $result = [
+            'start' => $deliveryExecution->getStartTime(),
+            'end' => $deliveryExecution->getFinishTime(),
+            'label' => $deliveryExecution->getLabel()
+        ];
+        return $result;
+    }
+
+
+    /**
+     * Get rubric to be printed
+     * Rubric is considered printed if it included to the section which has an item tagged by specified tag
+     * (@see )
+     *
+     * @param \taoDelivery_models_classes_execution_DeliveryExecution $deliveryExecution
+     */
+    public function getPrintableRubric(\taoDelivery_models_classes_execution_DeliveryExecution $deliveryExecution)
+    {
+        /** @var DeliveryService $deliveryService */
+        $deliveryService = $this->getServiceManager()->get('taoProctoring/delivery');
+        $session = $deliveryService->getTestSession($deliveryExecution);
+
+        $inputParameters = $deliveryService->getRuntimeInputParameters($deliveryExecution);
+        $testDefinition = \taoQtiTest_helpers_Utils::getTestDefinition($inputParameters['QtiTestCompilation']);
+        $sections = $testDefinition->getComponentsByClassName('assessmentSection');
+
+        $tag = $this->getOption(self::OPTION_PRINTABLE_RUBRIC_TAG);
+
+        $directoryIds = explode('|', $inputParameters['QtiTestCompilation']);
+        $compilationDirs = array(
+            'private' => \tao_models_classes_service_FileStorage::singleton()->getDirectoryById($directoryIds[0]),
+            'public' => \tao_models_classes_service_FileStorage::singleton()->getDirectoryById($directoryIds[1]),
+        );
+
+        $rubrics = [];
+
+        // -- variables used in the included rubric block templates.
+        // base path (base URI to be used for resource inclusion).
+        $basePathVarName = TAOQTITEST_BASE_PATH_NAME;
+        $$basePathVarName = $compilationDirs['public']->getPublicAccessUrl();
+
+        // state name (the variable to access to get the state of the assessmentTestSession).
+        $stateName = TAOQTITEST_RENDERING_STATE_NAME;
+        $$stateName = $session;
+
+        // views name (the variable to be accessed for the visibility of rubric blocks).
+        $viewsName = TAOQTITEST_VIEWS_NAME;
+        $$viewsName = array(View::CANDIDATE);
+
+        foreach ($sections as $section) {
+            $assessmentItemsRef = $section->getComponentsByClassName('assessmentItemRef');
+            foreach ($assessmentItemsRef as $item) {
+                foreach($item->getCategories() as $category) {
+                    if ($category === $tag) {
+                        foreach ($section->getRubricBlockRefs() as $rubric) {
+                            ob_start();
+                            include($compilationDirs['private']->getPath() . $rubric->getHref());
+                            $rubrics[] = ob_get_clean();
+                        }
```

Comment:
Actually I'm almost completely copied these lines from here:
https://github.com/oat-sa/extension-tao-testqti/blob/develop/helpers/class.TestRunnerUtils.php#L517-L539

I don't think that we can render it without using output buffering because these files (rubric templates) located in the private directory with compiled test.